src/compiler/chainrules.jl

```diff
@@ -110,3 +110,51 @@ As per [`chain_rrule`](@ref) but with support for kwargs.
   end
   return y, kw_zpullback
 end
+
+"""
+    rrule_via_ad(f, args)
+
+Function with the same API as the `ChainRulesCore.rrule`, used for testing Zygote gradients
+with `ChainRulesTestUtils.test_rrule` functionality.
+
+```
+ChainRulesTestUtils.test_rrule(round, 2.2; rrule_f=rrule_via_ad)
+```
+"""
+function rrule_via_ad(f::Function, args...)
+    y, pb = pullback(f, args...)
+    function ad_pullback(Δ)
+        return (NoTangent(), zygote2differential(pb(wrap_chainrules_output(Δ)), args)...)
+    end
+    return y, ad_pullback
+end
+
+"""
+    zygote2differential(x)
```

Comment:
This function is essentially a generalisation of `wrap_chainrules_input` that preserves the information about the primal type.

I am keeping it separate because it might eventually be moved to ZygoteRules when we support ChainRules types internally in Zygote.

As an aside: That PR is nearly complete (only higher order AD is broken) but 6 months old and would need a massive rebase. The hope is that after we have the possibility of rules for higher order functions by calling back into AD, the number of rules in Zygote would be much smaller and the PR would become easier.